### PR TITLE
[dashboard] BillingAccountSelector: Re-add filter for "subscription p…

### LINKED ISSUE #13734: Incorrect billing mode for teams without usage-based billing configured
Since #13461 was merged it appears that teams which have no billing configured are showing up in users's choice of "**Default Billing Account**".  e.g. In my case I see wow-team which was not there before and does not have UBP configured.

Screenshot 2022-10-10 at 22 25 14

### PR DESCRIPTION
**Pull request by geropl**
…resent"

## Description
image

## Related Issue(s)

Fixes #13734 

## How to test
 - go to [preview env](https://gpl-13734-bacc.preview.gitpod-dev.com/workspaces)
 - create three teams:
   1. `Gitpod-with-plan` UBP with plan
   2. `Gitpod-without-plan` UBP without plan
   3. `some-team` Chargebee
 - Go to [personal billing settings](https://gpl-13734-bacc.preview.gitpod-dev.com/billing) and notice you can only select:
   a. myself
   b. `Gitpod-with-plan`

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Comment by geropl** [comment]
rebased and waiting for preview env for testing

File: components/dashboard/src/components/BillingAccountSelector.tsx [hunk 1]
```diff
@@ -24,7 +24,19 @@ export function BillingAccountSelector(props: { onSelected?: () => void }) {
...
+        const teamsWithBilling: Team[] = [];
+        Promise.all(
+            teams.map(async (t) => {
+                const attributionId: string = AttributionId.render({ kind: "team", teamId: t.id });
+                const subscriptionId = await getGitpodService().server.findStripeSubscriptionId(attributionId);
+                if (subscriptionId) {
+                    teamsWithBilling.push(t);
+                }
+            }),
+        ).then(() => setTeamsAvailableForAttribution(teamsWithBilling.sort((a, b) => (a.name > b.name ? 1 : -1))));
```
Comment (easyCZ) [inline components/dashboard/src/components/BillingAccountSelector.tsx:38]: Does this need to be enforced server-side also?

File: components/dashboard/src/components/BillingAccountSelector.tsx [hunk 2]
```diff
@@ -24,7 +24,19 @@ export function BillingAccountSelector(props: { onSelected?: () => void }) {
...
             return;
         }
-        setTeamsAvailableForAttribution(teams.sort((a, b) => (a.name > b.name ? 1 : -1)));
+
+        // Filter teams based on whether they can actually "resolve" any credits thrown at them
+        const teamsWithBilling: Team[] = [];
+        Promise.all(
+            teams.map(async (t) => {
+                const attributionId: string = AttributionId.render({ kind: "team", teamId: t.id });
+                const subscriptionId = await getGitpodService().server.findStripeSubscriptionId(attributionId);
```
Comment (easyCZ) [inline components/dashboard/src/components/BillingAccountSelector.tsx:33]: Does this really need to actually search Stripe? Could we use the CostCenter instead for this?

**Comment by easyCZ** [comment]
I've cut the following for the follow-ups:
* https://github.com/gitpod-io/gitpod/issues/13977
* https://github.com/gitpod-io/gitpod/issues/13978

**Pull request merged**